pkg/synthetictests/duplicated_events.go

```diff
@@ -191,3 +214,85 @@ func getTimesAnEventHappened(message string) (string, int) {
 	}
 	return matches[0][1], int(times)
 }
+
+// isConsoleReadinessDuringInstallation returns true if the event is for console readiness and it happens during the
+// initial installation of the cluster.
+// we're looking for something like
+// > ns/openshift-console pod/console-7c6f797fd9-5m94j node/ip-10-0-158-106.us-west-2.compute.internal - reason/ProbeError Readiness probe error: Get "https://10.129.0.49:8443/health": dial tcp 10.129.0.49:8443: connect: connection refused
+// with a firstTimestamp before the cluster completed the initial installation
+func isConsoleReadinessDuringInstallation(monitorEvent monitorapi.EventInterval, kubeClientConfig *rest.Config) bool {
+	if !strings.Contains(monitorEvent.Locator, "ns/openshift-console") {
+		return false
+	}
+	if !strings.HasPrefix(monitorEvent.Locator, "pod/console-") {
+		return false
+	}
+	if !strings.Contains(monitorEvent.Locator, "Readiness probe") {
+		return false
+	}
+	if !strings.Contains(monitorEvent.Locator, "connect: connection refused") {
+		return false
+	}
+	tokens := strings.Split(monitorEvent.Locator, " ")
+	tokens = strings.Split(tokens[1], "/")
+	podName := tokens[1]
```

Comment:
if we're going to split into tokens, can we just make a `map[string][string]` instead of using the `Locator` `Contains` business above?  It is unlikely that `ns/openshift-console` is a substring of something in the `Locator` that is not the namespace entry, but it feels weird to do some unstructured checking before unmarshalling when we could unmarshal and then do structured checking.  And then here, when you just assume that the second token is the pod name, you could use `tokens["pod"]` to conveniently get that pod name without being as brittle.

Reply:
> if we're going to split into tokens, can we just make a `map[string][string]` instead of using the `Locator` `Contains` business above? It is unlikely that `ns/openshift-console` is a substring of something in the `Locator` that is not the namespace entry, but it feels weird to do some unstructured checking before unmarshalling when we could unmarshal and then do structured checking. And then here, when you just assume that the second token is the pod name, you could use `tokens["pod"]` to conveniently get that pod name without being as brittle.

This is a good general idea given the event compression format.  I think I'm going to try to move the other way and keep all the event content as a real event.